src/compiler/compile/Component.ts

```diff
@@ -1079,7 +1093,7 @@ export default class Component {
 						} else if (owner === instance_scope) {
 							const variable = var_lookup.get(name);
 
-							if (variable.reassigned || variable.mutated) hoistable = false;
+							if (variable.reassigned || variable.mutated || variable.aliased) hoistable = false;
```

Comment:
I don't think this is the right place for this. This is marking the function not hoistable if it references aliased variables, instead of marking the function not hoistable if itself is aliased.